### ISSUE TITLE
OMXPlayer not playing very short videos, <1 sec

### ISSUE THREAD

**Issue by dannibje** [popcornmix/omxplayer#520]

**Comment by jehutting** [comment]
@dannibje Did you tried it with  --audio_fifo, --video_fifo, --audio_queue and --video_queue options?
The smallest mp4 I have with a `Duration: 00:00:01.92` plays/loops fine.
Otherwise try to create a logging (omxplayer.log) with the `-g` option, or just post the video.

**Comment by jehutting** [comment]
@dannibje Thanks for the logging. Indeed, From it, I can see that the video is never started. It issues a `SubmitEOS`, but never catch this EOS (missing `IsEOS` in logging). It seems to stop (the rest of the logging after the `SubmitEOS`, I have to verify that for sure).

I can't determine the duration of the video.

Try to see what `--audio_fifo 0` and/or `--video_fifo 0` does.

Possible to post the video?

**Comment by dannibje** [comment]
@jehutting Thanks for your quick answers.  I tried playing the video with --audio_fifo 0 and/or --video_fifo 0 options and it seems to give the same results.  Sending the logs in case you can see something I cant. 
I attached the video so you can check it out.  I know it is an m4v file but I have had the same issue with mp4 and successfully played longer m4v videos with the omxplayer (The video plays but is not correctly enoded stuff).  


[omxLogs.zip](https://github.com/popcornmix/omxplayer/files/739292/omxLogs.zip)
[IMG_0884.m4v.zip](https://github.com/popcornmix/omxplayer/files/739225/IMG_0884.m4v.zip)

**Comment by jehutting** [comment]
Again a problem of the media time clock not getting to run.

It is the same as I found out [here](https://github.com/popcornmix/omxplayer/issues/297#issuecomment-68965612). With the [patch](https://github.com/popcornmix/omxplayer/issues/297#issuecomment-69431050) which forces the call to OMXVideo::PortSettingChanged, your short video plays...well that is to say... it is more a flash as it is so short :-).

Out of curiosity, what's the purpose of having so small videos?

**Comment by dannibje** [comment]
Thank you so much for your help!
And to satisfy your curiosity.  People can upload any kind of videos to our platform and those videos get displayed on a TV-screen via RPi.  Someone uploaded a very short video and the whole thing froze.  We fixed it with a timer that triggered after x-time.  But that is a hack, thank you for the fix!

**Comment by tvjon** [comment]
Actually, omxplayer will play it, but only via a contrived way...

 $ omxplayer -v
omxplayer - Commandline multimedia player for the Raspberry Pi
        Build date: Mon, 30 Jan 2017 01:52:37 +0000
        Version   : 62fb580 [master]
        Repository: https://github.com/popcornmix/omxplayer.git

$ omxplayer --display 4 IMG_0884.m4v 
Video codec omx-h264 width 1080 height 1920 profile 100 fps 24.000000
Audio codec aac channels 1 samplerate 44100 bitspersample 16
Subtitle count: 0, state: off, index: 1, delay: 0
Seek to: 00:00:00
V:PortSettingsChanged: 1080x1920@24.00 interlace:0 deinterlace:0 anaglyph:0 par:1.00 display:4 layer:0 alpha:255 aspectMode:0

At this stage, pressing left-arrow (quickly followed by pressing space if you want to study it) will show it. I also managed to get dbuscontrol.sh to do similar, but inconsistently, so possibly with the right combination (can dbus commands be concatenated?) it could be done properly.

![2017-03-03_13 10 17](https://cloud.githubusercontent.com/assets/5360158/23552605/afe4882c-0014-11e7-8fec-5d6a7f248ba0.jpg)

**Comment by rbckman** [comment]
Hello! I've been running into this same issue. Is there a workaround / fix? I'm building a camera that can glue your film together. It would be important to get to see your takes even if they're just around 2 secs. But at the moment the playback only works 2 out of 10 times due to this issue. :'(

heres the log [omxplayer.log](https://github.com/popcornmix/omxplayer/files/3104137/omxplayer.log)
ps. heres a link to my camera project called [tarina](https://github.com/rbckman/tarina)

**Comment by rbckman** [comment]
hmm.. wondering if this [pull request](https://github.com/popcornmix/omxplayer/pull/669) would fix this issue.

**Comment by dronus** [comment]
I don't think that would help. There where some discussions (can't find them now) for this well known bug of omxplayer to not play very short files. The bug as I remember was inside the firmware / hardware stack for video decoding (OpenMAX) and there whas no idea how omxplayer could reliable circumvent the problem. Maybe that is fixed as of today, but maybe not. If not, pausing or resuming the video won't help, as it is the software / hardware of the accelerated video codec that won't start running until at least x kb or frames are fed into.

**Comment by rbckman** [comment]
thanks! figured out how to do it! tvjon, you said the video will play if you press left key (previous video). So im using omxplayer through Will Prices wrapper https://github.com/willprice/python-omxplayer-wrapper and it's as easy as to send the player.previous() command before you start the video and it works!
